home-manager/neovim/default.nix

```diff
@@ -49,6 +54,23 @@ in
     pkgs.vimPlugins.foreign-heirline-components
     heirline-nvim
 
-    vim-nix
+    mason-nvim
+    mason-lspconfig-nvim
+    mason-tool-installer-nvim
+    cmp-nvim-lsp
+    fidget-nvim
+    pkgs.vimPlugins.foreign-lazydev-nvim
+    pkgs.vimPlugins.foreign-luvit-meta
+    nvim-lspconfig
+
+    conform-nvim
+
+    cmp-path
+    cmp-nvim-lsp
+    cmp_luasnip
+    luasnip
+    nvim-cmp
   ];
+
+  extraLuaPackages = ps: [ ps.jsregexp ];
```

Comment:
Needed as a build step for `luasnip`. Blocked until I came across this [issue](https://github.com/NixOS/nixpkgs/issues/306367). 

```lua
      {
        'L3MON4D3/LuaSnip',
        build = (function()
          -- Build Step is needed for regex support in snippets.
          -- This step is not supported in many windows environments.
          -- Remove the below condition to re-enable on windows.
          if vim.fn.has 'win32' == 1 or vim.fn.executable 'make' == 0 then
            return
          end
          return 'make install_jsregexp' <------ Stuck thinking how do I do this in nix
        end)(),
      }
```